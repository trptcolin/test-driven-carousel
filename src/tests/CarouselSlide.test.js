import React from "react";
import { shallow, mount } from "enzyme";
import CarouselSlide from "../CarouselSlide";
import styled from "styled-components";

describe("Img", () => {
  let mounted;
  const imgUrl = "https://example.com/default.jpg";

  beforeEach(() => {
    const Img = CarouselSlide.defaultProps.Img;
    mounted = mount(<Img src={imgUrl} imgHeight={500} />);
  });

  it("uses imgHeight as the height style property", () => {
    expect(mounted).toHaveStyleRule("height", "500px");
    mounted.setProps({ imgHeight: "calc(100vh - 100px" });
    expect(mounted).toHaveStyleRule("height", "calc(100vh - 100px");
  });

  it("renders correctly", () => {
    expect(mounted).toMatchSnapshot();
  });
});

describe("CarouselSlide", () => {
  let wrapper;

  const defaultImgUrl = "https://examples.com/image.png";
  const defaultDescription = "Super sweet photo";

  beforeEach(() => {
    wrapper = shallow(
      <CarouselSlide imgUrl={defaultImgUrl} description={defaultDescription} />
    );
  });

  it("passes other props through to the figure", () => {
    const style = {};
    const onClick = () => {};
    const className = "my-carousel-slide";
    wrapper.setProps({ style, onClick, className });
    expect(wrapper.prop("style")).toBe(style);
    expect(wrapper.prop("onClick")).toBe(onClick);
    expect(wrapper.prop("className")).toBe(className);
  });

  it("renders correctly", () => {
    wrapper.setProps({
      description: "description",
      attribution: "attribution",
    });
    expect(wrapper).toMatchSnapshot();
  });
});
