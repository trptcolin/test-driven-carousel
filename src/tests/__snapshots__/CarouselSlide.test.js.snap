// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselSlide renders correctly 1`] = `
<figure>
  <CarouselSlide__DefaultImg
    imgHeight={500}
    src="https://examples.com/image.png"
  />
  <figcaption>
    <strong>
      description
    </strong>
     
    attribution
  </figcaption>
</figure>
`;

exports[`Img renders correctly 1`] = `
.c0 {
  object-fit: cover;
  width: 100%;
  height: 500px;
}

<CarouselSlide__DefaultImg
  imgHeight={500}
  src="https://example.com/default.jpg"
>
  <StyledComponent
    forwardedClass={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "CarouselSlide__DefaultImg-ds4ujd-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "object-fit:cover;width:100%;height:",
            [Function],
            ";",
          ],
        },
        "displayName": "CarouselSlide__DefaultImg",
        "render": [Function],
        "styledComponentId": "CarouselSlide__DefaultImg-ds4ujd-0",
        "target": "img",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    imgHeight={500}
    src="https://example.com/default.jpg"
  >
    <img
      className="c0"
      src="https://example.com/default.jpg"
    />
  </StyledComponent>
</CarouselSlide__DefaultImg>
`;
